enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1273 @@
+1. SRE can address the issue with a system configuration file applied in a MachineConfig.
+1. SRE creates the MachineConfig for the customer and provides the customer the option to either (a) wait until their
+   configured maintenance schedule permits the material change from being initiated by the MachineConfigOperator
+   or (b) modify change management to permit immediate application (e.g. setting `disabledUntil`).   
+1. The problem is not pervasive, so the customer chooses the deferred remediation. 
+1. The change is initiated and nodes are rebooted during the next permissive window.
+
+
+#### On-prem Standalone GitOps Change Management Scenario
+1. An on-prem cluster is fully managed by gitops. As changes are committed to git, those changes are applied to cluster resources.
+1. Configurable stanzas of the ClusterVersion and MachineConfigPool(s) resources are checked into git.
+1. The cluster lifecycle administrator configures `changeManagement` in both the ClusterVersion and worker MachineConfigPool
+   in git. The MaintenanceSchedule strategy is chosen. The policy permits control-plane and worker-node updates only after
+   19:00 Eastern US.
+1. During the working day, users may contribute and merge changes to MachineConfigs or even the `desiredUpdate` of the
+   ClusterVersion. These resources will be updated in a timeline manner via GitOps.
+1. Despite the resource changes, neither the CVO nor MCO will begin to initiate the material changes on the cluster.
+1. Privileged users who may be curious as to the discrepancy between git and the cluster state can use `oc get -o=yaml/describe`
+   on the resources. They observe that changes are pending and the time at which changes will be initiated.
+1. At 19:00 Eastern, the pending changes begin to be initiated. This rollout abides by documented OpenShift constraints
+   such as the MachineConfigPool `maxUnavailable` setting.
+   
+#### On-prem Standalone Manual Strategy Scenario
+1. A small, business critical cluster is being run on-prem.
+1. There are no reoccurring windows of time when the cluster lifecycle administrator can tolerate downtime.
+   Instead, updates are negotiated and planned far in advance. 
+1. The cluster workloads are not HA and unplanned drains are considered a business risk.
+1. To prevent surprises, the cluster lifecycle administrator sets the Manual strategy on the worker MCP.
+1. Given the sensitivity of the operation, the lifecycle administrator wants to manually drain and reboot
+   nodes to accomplish the update.
+1. The cluster lifecycle administrator sends a company-wide notice about the period during which service may be disrupted.
+1. The user determines the most recent rendered worker configuration. They configure the `manual` change
+   management policy to use that exact configuration as the `desiredConfig`. 
+1. The MCO is thus being asked to ignite any new node or rebooted node with the desired configuration, but it

```

Comment:
The MCO logic today dictates that there must be at least 1 node that's upgraded to the latest version, before any new nodes will ignite into it directly. I assume that needs to change for this?

Reply:
For both manual and assisted, the proposal would say "desired config (version)" (as declared in the MCP) vs "latest version" (whatever might have rendered most recently).
If we apply the question to the "desired config", I think the **manual** strategy would indeed change that behavior (new nodes would ignite with the desired config even if they were the first. That ensures that the administrator has a fixed set of nodes they need to manually reboot after making the MCP change.
For the **assisted** strategy, I would be OK with maintaining the current behavior. Since it is a semi-automated process, the administrator would likely not even notice if a new node came up within the validation window. I'll add notes on this.